From 04361071284d5f2cffe6b59ad86422b02d186f77 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Mon, 8 Sep 2014 15:07:23 -0400
Subject: [ethernet] ixgbevf: have clean_rx_irq return total_rx_packets cleaned

Message-id: <1410188883-6595-9-git-send-email-jogreene@redhat.com>
Patchwork-id: 92357
O-Subject: [RHEL7.1 PATCH 08/48] ixgbevf: have clean_rx_irq return total_rx_packets cleaned
Bugzilla: 1091124
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: 1091124
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

Rather than return true/false indicating whether there was budget left, return
the total packets cleaned. This currently has no use, but will be used in a
following patch which enables CONFIG_NET_RX_BUSY_POLL support in order to track
how many packets were cleaned during the busy poll as part of the extended
statistics.

Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 08e50a20ed05fba11c7dbc9e325369bef6a1c194)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
index 5963ea4..1402287 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -410,9 +410,9 @@ static inline void ixgbevf_irq_enable_queues(struct ixgbevf_adapter *adapter,
  IXGBE_WRITE_REG(hw, IXGBE_VTEIMS, qmask);
 }
 
-static bool ixgbevf_clean_rx_irq(struct ixgbevf_q_vector *q_vector,
-     struct ixgbevf_ring *rx_ring,
-     int budget)
+static int ixgbevf_clean_rx_irq(struct ixgbevf_q_vector *q_vector,
+    struct ixgbevf_ring *rx_ring,
+    int budget)
 {
  struct ixgbevf_adapter *adapter = q_vector->adapter;
  struct pci_dev *pdev = adapter->pdev;
@@ -540,7 +540,7 @@ next_desc:
  q_vector->rx.total_packets += total_rx_packets;
  q_vector->rx.total_bytes += total_rx_bytes;
 
- return !!budget;
+ return total_rx_packets;
 }
 
 /**
@@ -572,8 +572,9 @@ static int ixgbevf_poll(struct napi_struct *napi, int budget)
 
  adapter->flags |= IXGBE_FLAG_IN_NETPOLL;
  ixgbevf_for_each_ring(ring, q_vector->rx)
-  clean_complete &= ixgbevf_clean_rx_irq(q_vector, ring,
-             per_ring_budget);
+  clean_complete &= (ixgbevf_clean_rx_irq(q_vector, ring,
+       per_ring_budget)
+       < per_ring_budget);
  adapter->flags &= ~IXGBE_FLAG_IN_NETPOLL;
 
  /* If all work not completed, return budget and keep polling */
-- 
1.7.1